ur_controllers/src/gpio_controller.cpp

```diff
@@ -446,9 +446,9 @@ bool GPIOController::handBackControl(std_srvs::srv::Trigger::Request::SharedPtr
   // call the service in the hardware
   command_interfaces_[CommandInterfaces::HAND_BACK_CONTROL_CMD].set_value(1.0);
 
-  while (command_interfaces_[CommandInterfaces::HAND_BACK_CONTROL_ASYNC_SUCCESS].get_value() == ASYNC_WAITING) {
-    // Asynchronous wait until the command has been executed
-    std::this_thread::sleep_for(std::chrono::milliseconds(50));
+  if(!waitForAsyncCommand([&](){return command_interfaces_[CommandInterfaces::IO_ASYNC_SUCCESS].get_value();}))
+  {
+    RCLCPP_WARN(get_node()->get_logger(),"Could not verify that io was set. (This might happen when using the mocked interface)");
```

Comment:
```suggestion
  if(!waitForAsyncCommand([&](){return command_interfaces_[CommandInterfaces::HAND_BACK_CONTROL_ASYNC_SUCCESS].get_value();}))
  {
    RCLCPP_WARN(get_node()->get_logger(),"Could not verify that hand_back_control was correctly triggered. (This might happen when using the mocked interface)");
```